contracts/liquidity_hub/pool-network/terraswap_pair/src/commands.rs

```diff
@@ -470,8 +470,11 @@ pub fn swap(
     }
 
     #[cfg(feature = "osmosis")]
-    if !swap_computation.osmosis_fee_amount.is_zero() {
-        // send osmosis fee to the Community Pool
+    if !swap_computation.osmosis_fee_amount.is_zero()
+        && info.sender != config.cosmwasm_pool_interface
```

Comment:
if the info sender is not the pool interface contract, i.e. not executed via the osmosis pool manager, then pay the fee to the community pool.